### PR TITLE
Feature/stepper

### PR DESCRIPTION
**Pull request by ydomenjoud**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <kal-panel title="organisation" icon="building">
+        lorem
+    </kal-panel>
+    <kal-panel title="preferences" icon="sliders">
+        lorem
+    </kal-panel>
+</kal-stepper>
+```
+
+* Doit afficher la liste d'étape donnée
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/README.md]: French ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KalStepHeaderComponent implements OnDestroy {
+  /** State of the given step. */
+  @Input() state: string;
+
+  /** Label of the given step. */
+  @Input() label: KalStepLabelDirective | string;
+
+
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: Remove space

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  templateLabel(): KalStepLabelDirective | null {
+    return this.label instanceof KalStepLabelDirective ? this.label : null;
+  }
+
+  /** Returns the host HTML element. */
+  getHostElement() {
+    return this._element.nativeElement;
+  }
+
+
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: Remove space

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  /** Returns MatStepLabel if the label of given step is a template label. */
+  templateLabel(): KalStepLabelDirective | null {
+    return this.label instanceof KalStepLabelDirective ? this.label : null;
+  }
+
+  /** Returns the host HTML element. */
+  getHostElement() {
+    return this._element.nativeElement;
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: We have not defined as convention prefix  '_' for private variables; 
Should we?
Currently it is not

Comment (ydomenjoud) [reply projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: this is inherited from CDK ...

Comment (Bellico) [reply projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: What  ?  (I speak for everything)

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  @Input() selected: boolean;
+
+  /** Whether the given step label is active. */
+  @Input() active: boolean;
+
+  /** Whether the given step is optional. */
+  @Input() optional: boolean;
+
+  constructor(
+    private _focusMonitor: FocusMonitor,
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: change private properties label

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    return this._element.nativeElement;
+  }
+
+
+  focus() {
+    this.getHostElement().focus();
+  }
+
+  ngOnDestroy() {
+    this._focusMonitor.stopMonitoring(this._element.nativeElement);
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts]: change private properties label

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.component.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  HostBinding, Input,
+  QueryList,
+  ViewEncapsulation
+} from '@angular/core';
+import { CdkStepper } from '@angular/cdk/stepper';
+import { FocusableOption } from '@angular/cdk/a11y';
+import { takeUntil } from 'rxjs/operators';
+
+import { KalStepComponent } from './kal-step/kal-step.component';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.component.ts]: imports

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,127 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component, NO_ERRORS_SCHEMA, ViewChild } from '@angular/core';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+
+import { KalStepperComponent } from './kal-stepper.component';
+import { KalStepComponent } from './kal-step/kal-step.component';
+import { KalStepLabelDirective } from './directives/kal-step-label.directive';
+import { KalStepHeaderComponent } from './kal-step-header/kal-step-header.component';
+import { KalStepperModule } from 'projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.module';
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.component.spec.ts]: imports

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

**Review by MichaelGrosio — APPROVED** [review]

**Pull request merged**